plugins/BEdita/API/src/Controller/LoginController.php

```diff
@@ -39,34 +39,45 @@ public function initialize()
         parent::initialize();
 
         if ($this->request->getParam('action') === 'login') {
-            $this->Auth->setConfig(
-                'authenticate',
-                [
-                    AuthComponent::ALL => [
-                        'scope' => [
-                            'blocked' => false,
-                        ],
-                        'contain' => ['Roles'],
+            $authenticationComponents = [
+                AuthComponent::ALL => [
+                    'scope' => [
+                        'blocked' => false,
                     ],
-                    'Form' => [
-                        'fields' => [
-                            'username' => 'username',
-                            'password' => 'password_hash',
-                        ],
-                        'passwordHasher' => [
-                            'className' => 'Fallback',
-                            'hashers' => [
-                                'Default',
-                                'Weak' => ['hashType' => 'md5'],
-                            ],
-                        ],
+                    'contain' => ['Roles'],
+                ],
+                'Form' => [
+                    'fields' => [
+                        'username' => 'username',
+                        'password' => 'password_hash',
                     ],
-                    'BEdita/API.Jwt' => [
-                        'queryDatasource' => true,
+                    'passwordHasher' => [
+                        'className' => 'Fallback',
+                        'hashers' => [
+                            'Default',
+                            'Weak' => ['hashType' => 'md5'],
+                        ],
                     ],
                 ],
-                false
-            );
+                'BEdita/API.Jwt' => [
+                    'queryDatasource' => true,
+                ],
+            ];
+
+            // TODO: better way to dynamically add external authentication components?
+            $uuidProvider = TableRegistry::get('AuthProviders')->find()->where(['name' => 'uuid'])->first();
```

Comment:
Agreed, it's problematic. We should have no knowledge of what type of providers are available.
Why not load **ALL** `auth_providers` ? 
Problem remains with configuration that may vary between providers. 
Why not impose a fixed configuration like this for now and then handle dynamic configurations as we create a new  `authProvider`?